deploy/helm/cht-chart/templates/eks-api-ingress.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if eq .Values.environment "remote" }}
+
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: api-ingress
+  annotations:
```

Comment:
While it's likely that that this PR will be run against EKS for the foreseeable future, there may be folks who may want to run it on GKS or other k8s providers. And the annotations here, if I recall correctly, would only be relevant for ALBs. That would include generation of certs. Would it be worth:

1. Externalizing these annotations and the desired ingress controller type into `Values.yaml` and have them pulled in here with an {{ if eq .Values.ingress.type }} or something similar.
2. Support other ways of auto-generating certs and renewing them. I think this PR uses a pre-generated cert. LE certs, because of their relatively short expiry windows, for instance, may not be too amenable to this. Here, I am thinking of adding the capability to support something like `cert-manager` and it's `cluster-issuer`.

Reply:
The first comment is addressed. Check out the `.Values.cluster_type` Different ingress types will be added and executed based on that parameter. Right now we support traefik (for k3s-k3d) and alb type for eks.
Re: the second comment: there have been issues trying to manage certs on our own in the past and decided to not do them. However, I think with k8s it's worth trying. I'd say for another PR though :)